src/Playwright/Core/Waiter.cs

```diff
@@ -218,6 +235,13 @@ internal async Task<T> WaitForPromiseAsync<T>(Task<T> task, Action dispose = nul
             Dispose();
             throw new TimeoutException(ex.Message + FormatLogRecording(_logs), ex);
         }
+        catch (OperationCanceledException ex)
+        {
+            dispose?.Invoke();
+            _error = ex.ToString();
+            Dispose();
+            throw;
```

Comment:
would be nice to throw the same exception but with formatted logs like in line 250.

Reply:
With `OperationCanceledException` the suggested flow is to throw the original exception to have better understanding of the place of cancelation. https://devblogs.microsoft.com/premier-developer/recommended-patterns-for-cancellationtoken/